metrics/metrics/src/main/java/io/helidon/metrics/HelidonGauge.java

```diff
@@ -19,42 +19,198 @@
 import java.util.Objects;
 import java.util.function.Function;
 import java.util.function.Supplier;
+import java.util.function.ToDoubleFunction;
 
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Metrics;
 import org.eclipse.microprofile.metrics.Gauge;
 import org.eclipse.microprofile.metrics.Metadata;
+import org.eclipse.microprofile.metrics.Tag;
 
 /**
  * Gauge implementation.
  */
-final class HelidonGauge<T extends Number> extends MetricImpl implements Gauge<T> {
-    private final Supplier<T> value;
+abstract class HelidonGauge<N extends Number> extends MetricImpl implements Gauge<N> {
+    /*
+     * The MicroProfile metrics API parameterizes its gauge type as Gauge<N extends Number> which is the type of
+     * value the gauge reports via its getValue() method. To register a gauge, the developer passes us a function-plus-target or
+     * a supplier with the return value from the function or supplier similarly parameterized with the subtype of Number.
+     *
+     * On the other hand, each Micrometer gauge is not parameterized and reports a double value.
+     *
+     * As a result, we do not have what we need to (easily) instantiate the correct subtype of Number, set its value based on
+     * the Micrometer delegate's value() result, and return the correctly-typed and -assigned value from our getValue() method.
+     *
+     * To work around this, we keep track ourselves of the function and target or supplier which report the gauge value.
+     * Then our getValue() implementation simply invokes the function on the target or the supplier rather than delegating to
+     * the Micrometer gauge value() method (which would do exactly the same thing anyway).
+     *
+     * This way the typing works out (with the expected unchecked cast).
+     */
 
-    private HelidonGauge(String registryType, Metadata metadata, Gauge<T> metric) {
-        super(registryType, metadata);
+    private final io.micrometer.core.instrument.Gauge delegate;
```

Comment:
I thought we should have support for Micrometer and possibly other metrics implementation, but not hardcoded into our impl, but through  provider modules.
Why is there a hard dependency on Micrometer?

Reply:
See earlier note about the neutral API and timing.

Reply:
I did not realize from the description that we must use Micrometer to be able to generate prometheus output (as we used to generate it ourself).
I have approved the PR, we need to fix this eventually when we switch to Helidon Metrics API

Reply:
Given that our neutral API will layer on one or more actual implementations (Micrometer, OTel metrics), we should delegate to the underlying implementation to publish, whether Prometheus or other (push or pull) models, rather than duplicate that functionality in our own code.